### PR TITLE
[AIRFLOW-5502][depends on AIRFLOW-5498, AIRFLOW-5499] Move GCP base hook to core

### PR DESCRIPTION
**Pull request by turbaszek**
Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-5502

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

**Comment by turbaszek** [comment]
Depends on #6122 #6121

File: airflow/contrib/hooks/gcp_api_base_hook.py [hunk 1]
```diff
@@ -16,299 +16,14 @@
...
-                finally:
-                    if current_env_state is None:
-                        if CREDENTIALS in os.environ:
-                            del os.environ[CREDENTIALS]
-                    else:
-                        os.environ[CREDENTIALS] = current_env_state
-        return wrapper
+warnings.warn(
+    "This module is deprecated. Please use `airflow.gcp.hooks.base`.",
+    DeprecationWarning,
```
Comment (mik-laj) [inline airflow/contrib/hooks/gcp_api_base_hook.py]: Can you add stacklevel here?

File: airflow/contrib/hooks/google_discovery_api_hook.py [hunk 2]
```diff
@@ -16,132 +16,14 @@
...
-            else:
-                google_api_conn_client = getattr(google_api_conn_client, sub_function + '_next')
-                google_api_conn_client = google_api_conn_client(api_endpoint_instance, api_response)
+# pylint: disable=unused-import
+from airflow.gcp.hooks.discovery_api import GoogleDiscoveryApiHook  # noqa
 
-        return google_api_conn_client
+warnings.warn(
+    "This module is deprecated. Please use `airflow.gcp.hooks.discovery_api`.",
+    DeprecationWarning,
```
Comment (mik-laj) [inline airflow/contrib/hooks/google_discovery_api_hook.py]: Can you add stacklevel here?

File: airflow/contrib/operators/google_api_to_s3_transfer.py [hunk 3]
```diff
@@ -16,162 +16,14 @@
...
+from airflow.operators.google_api_to_s3_transfer import GoogleApiToS3Transfer  # noqa
 
-    def _expose_google_api_response_via_xcom(self, task_instance, data):
-        if sys.getsizeof(data) < MAX_XCOM_SIZE:
-            task_instance.xcom_push(key=self.google_api_response_via_xcom, value=data)
-        else:
-            raise RuntimeError('The size of the downloaded data is too large to push to XCom!')
+warnings.warn(
+    "This module is deprecated. Please use `airflow.operators.google_api_to_s3_transfer`.",
+    DeprecationWarning,
```
Comment (mik-laj) [inline airflow/contrib/operators/google_api_to_s3_transfer.py]: Can you add stacklevel here?

File: airflow/contrib/utils/gcp_field_sanitizer.py [hunk 4]
```diff
@@ -16,150 +16,16 @@
...
 
-    def sanitize(self, body):
-        """
-        Sanitizes the body according to specification.
-        """
-        for elem in self._sanitize_specs:
-            self._sanitize(body, elem, "")
+warnings.warn(
+    "This module is deprecated. Please use `airflow.gcp.utils.field_sanitizer`.",
+    DeprecationWarning
```
Comment (mik-laj) [inline airflow/contrib/utils/gcp_field_sanitizer.py]: Can you add stacklevel here?

File: airflow/contrib/utils/gcp_field_validator.py [hunk 5]
```diff
@@ -16,417 +16,20 @@
...
-                    "The field '%s' is in the body, but is not specified in the "
-                    "validation specification '%s'. "
-                    "This might be because you are using newer API version and "
-                    "new field names defined for that version. Then the warning "
-                    "can be safely ignored, or you might want to upgrade the operator"
-                    "to the version that supports the new API version.",
-                    field_name, self._validation_specs)
+warnings.warn(
+    "This module is deprecated. Please use `airflow.gcp.utils.field_validator`.",
+    DeprecationWarning
```
Comment (mik-laj) [inline airflow/contrib/utils/gcp_field_validator.py]: Can you add stacklevel here?

**Review by potiuk — APPROVED** [review]

**Pull request merged**